### ISSUE TITLE
[Feature Request] Allow X-value to be a non-time metric

### ISSUE THREAD

**Issue by Marenz** [grafana/grafana#5641]
* **I'm submitting a ...**
- [x] Feature request

Please include this information:
- What Grafana version are you using?
 - 3.0.4
- What datasource are you using?
 - graphite
- What OS are you running grafana on?
 - debian 8.0

I would love to see support for graphs that can have a normal data source metric as value for the X-axis. 
For example, assuming I have the two metrics `server.users_matches` and `server.matches_user`
I could draw them as such:

![image](https://cloud.githubusercontent.com/assets/424752/17062575/bc8849b0-5033-11e6-9e9c-45b56113dc5d.png)

The time info would be lost here of course, so we would display for example the current value or the avg or max or whatever you configure. Extra nice would be, of course, if you could pick the time you want to look at.

**Comment by Dieterbe** [comment]
basically it looks like you just want a histogram, for which there is already a ticket somewhere. and also have a look at https://grafana.net/plugins/mtanda-histogram-panel

**Comment by Marenz** [comment]
I checked out the histogram panel before creating this ticket, but from the description, I was given the impression that it has a rather specific use case:

> This plugin receives raw time series data, and count each value occurrence, and then show the occurrence as histogram.

What I have are 2 series of data one representing X, the other the Y values.

**Comment by torkelo** [comment]
strangely I could not find an existing issue for this! This is definitely something I hope to do soon!

**Comment by PaulKuiper** [comment]
HiTorkelo. There have been earlier request for this, but they are labeled as scatterplot I think. I'm pleased to inform you that Grafana has become a verb in our company (>700 users at the moment). Brillant piece of software and really good focus on UX. Thanks for your incredible hard work.

**Comment by torkelo** [comment]
closing this in favor or supporting histograms issue
https://github.com/grafana/grafana/issues/600

**Comment by PaulKuiper** [comment]
This issue has no relation to histograms whatsoever. It describes the situation in which you have two or more time series: t, y1.  t,y2. 

The request is not to plot t,y1 and t,y2 but to plot y1,y2 (the correlation) where y1 is chosen to be the x axis in this case. Only y values falling in the selected time range are plotted.

**Comment by torkelo** [comment]
@PaulKuiper Not sure how the correlation would work, it would be a correlation in time, could you give example with the actual data?

**Comment by PaulKuiper** [comment]
Hi Torkelo,

Sure, I copied in an image of a dashboard were we have such a correlation plot (we created our own panel using plot.ly to make the scatter plots):

![example](https://cloud.githubusercontent.com/assets/5039017/17658498/64bbc1f2-62cc-11e6-9e6f-286234ce71c7.png)

The top graph shows two signals as you would normally plot in Grafana (t,y1) and (t,y2). The bottom graph shows you the data we really want (y1,y2). It nicely shows how y2 depends on y1 over that same time interval. We have many examples of  these kind of correlation, because our application domain is process control. This is probably different from the normal Server/Network monitoring applications.

I could image the user interface options, which would appear if more than one metric is available:

![ui](https://cloud.githubusercontent.com/assets/5039017/17658694/c0368d04-62cd-11e6-9c34-969e4177fa41.png)

One of the signals becomes the x axis. In table form it looks as follows:

![ty1y2](https://cloud.githubusercontent.com/assets/5039017/17658996/e823271c-62cf-11e6-9bf5-baf29b8574e9.PNG)

The (y1, y2) point pairs can be plotted in a scatter plot with y1 being the x axis (or y2, whatever you select). The x axis would have the value range of the y1 signal. If you choose a larger time range, more or less points will be plotted. To prevent that unaligned timestamps cause the few points to align you can consider to interpolate (forward fill or linear interpolation) the x axis. It would result in the following table:

![ty1y2_interpolate](https://cloud.githubusercontent.com/assets/5039017/17658916/3146d34a-62cf-11e6-8b25-7b2fbd522ffb.PNG)

Now more (y1,y2) pairs are available for plotting. I guess it would be better to make a different kind of panel type for these kind of graphs (called scatter plots with a natural extension to heatmaps). I hope this clarified it a bit. Kind regards, 

Paul Kuiper

**Comment by hl68fx** [comment]
I would like to be able to change the x-axis to a column of my database. For example column stress for the y-axis and column strain for the x-axis. For example something like this: [https://www.researchgate.net/profile/Mehdi_Ghandil/publication/283665206/figure/fig12/AS:294511428943877@1447228404050/Fig-12-Stress-strain-diagram-for-the-mild-steel.png](url)

Just like Paul Kuiper mentioned, but with connected data points.
I think this would enlarge the area of application for Grafana a lot! :)

Kind regards,
Tom

**Comment by majj** [comment]
+1

**Comment by Marenz** [comment]
Is there any plan to reopen this ticket given that it was apparently closed based on the wrong assumption it is equal to histograms?

**Comment by samhatchett** [comment]
This issue may end up being a duplicate of #202 if I'm reading this correctly. We are talking about plotting two series against each other with time as a non-plotted parameter.

**Comment by ryantxu** [comment]
FYI, Check https://github.com/NatelEnergy/grafana-plotly-panel and https://grafana.com/plugins/natel-plotly-panel

**Comment by mohanraj1939** [comment]
+1 this for simple json datasource
